torch/csrc/distributed/c10d/Utils.hpp

```diff
@@ -675,7 +679,11 @@ void recvBytes(int socket, T* buffer, size_t length) {
     SYSCHECK_ERR_RETURN_NEG1(
         bytesReceived = recv(socket, currentBytes, bytesToReceive, 0))
     if (bytesReceived == 0) {
-      C10_THROW_ERROR(DistNetworkError, "failed to recv, got 0 bytes");
+      C10_THROW_ERROR(
+          DistNetworkError,
+          "Failed to recv, got 0 bytes. "
+          "Connection was likely closed. "
+          "Did the remote server shutdown or crash?");
```

Comment:
In the case of server normal shutdown vs. crash, will the return value/error code be different?

Reply:
In a normal shutdown we don't hit this path since there's no inflight operations. If the server shutsdown prematurely or crashing the return is effectively the same message. Typically this either shows up as this failed to recv or as a connection reset error depending on what operation was running.

We don't have way to distinguish between the two. Some apps do this at the protocol level by sending a "shutdown" message prior to closing but we can't do that without changing the protocol here